traja/models/train.py

```diff
@@ -0,0 +1,696 @@
+        lr_factor: float = 0.1,
+        scheduler_patience: int = 10,
+    ):
+
+        assert (
+            model_type in HybridTrainer.valid_models
+        ), "Model type is {model_type}, valid models are {}".format(
+            HybridTrainer.valid_models
+        )
+
+        self.model_type = model_type
+        self.input_size = input_size
+        self.lstm_hidden_size = lstm_hidden_size
+        self.num_lstm_layers = num_lstm_layers
+        self.classifier_hidden_size = classifier_hidden_size
+        self.num_classifier_layers = num_classifier_layers
+        self.batch_first = batch_first
+        self.reset_state = reset_state
+        self.output_size = output_size
+        self.num_classes = num_classes
+        self.latent_size = latent_size
+        self.num_classifier_layers = num_classifier_layers
+        self.num_future = num_future
+        self.epochs = epochs
+        self.batch_size = batch_size
+        self.num_past = num_past
+        self.dropout = dropout
+        self.bidirectional = bidirectional
+        self.loss_type = loss_type
+        self.optimizer_type = optimizer_type
+        self.lr = lr
+        self.lr_factor = lr_factor
+        self.scheduler_patience = scheduler_patience
+
+        self.model_hyperparameters = {
+            "input_size": self.input_size,
+            "num_past": self.num_past,
+            "batch_size": self.batch_size,
+            "lstm_hidden_size": self.lstm_hidden_size,
+            "num_lstm_layers": self.num_lstm_layers,
+            "classifier_hidden_size": self.classifier_hidden_size,
+            "num_classifier_layers": self.num_classifier_layers,
+            "num_future": self.num_future,
+            "latent_size": self.latent_size,
+            "output_size": self.output_size,
+            "num_classes": self.num_classes,
+            "batch_first": self.batch_first,
+            "reset_state": self.reset_state,
+            "bidirectional": self.bidirectional,
+            "dropout": self.dropout,
+        }
+
+        # Instantiate model instance based on model_type
+        if self.model_type == "ae":
+            self.model = MultiModelAE(**self.model_hyperparameters)
+        elif self.model_type == "vae":
+            self.model = MultiModelVAE(**self.model_hyperparameters)
+
+        # Classification task check
+        self.classify = True if self.classifier_hidden_size is not None else False
+
+        # Model optimizer and the learning rate scheduler
+        optimizer = Optimizer(
+            self.model_type, self.model, self.optimizer_type, classify=self.classify
+        )
+
+        self.model_optimizers = optimizer.get_optimizers(lr=self.lr)
+        self.model_lrschedulers = optimizer.get_lrschedulers(
+            factor=self.lr_factor, patience=self.scheduler_patience
+        )
+
+    def __str__(self):
+        return f"Training model type {self.model_type}"
+
+    def fit(self, train_loader, test_loader, model_save_path=None):
```

Comment:
We should make epochs an argument to fit(). You might want to change it on the fly.

Also, I think we should inject the training mode. You sometimes want to alternate and experiment with different training strategies for the heads.

I created a pull request to that effect